components/script/dom/htmlformelement.rs

```diff
@@ -974,7 +974,15 @@ pub trait FormControl: DomObject {
             elem.is_in_same_home_subtree(&*form)
         });
 
-        self.form_attribute_mutated(AttributeMutation::Removed);
+        // FIXME: We should call `form_attribute_mutated` method here,
+        // but also we don't want to call reset_form_owner() method in
+        // `form_attribute_mutated`. We should fix this.
+        // self.form_attribute_mutated(AttributeMutation::Removed);
```

Comment:
Let's split form_attribute_mutated into `register_if_necessary`, `unregister_if_necessary`, and `form_attribute_mutated`. `form_attribute_mutated` can call the other two methods based on the mutation, and code like `bind_form_control_to_tree` can call those methods and reset_form_owner as needed.